lib/saml_idp/request.rb

```diff
@@ -85,6 +85,11 @@ def log(msg)
     end
 
     def valid?
+      if issuer.nil?
+        log "Null issuer"
+        return false
+      end
```

Comment:
I think this the correct fix is to make sure that the "service_provider?" method below returns false, which mean the unless clause below will handle this case.

The problem is that @_service_provider is always getting set to something that claims to be valid? even when no service provider is found.     I think  :)